x-pack/plugins/lens/public/indexpattern_datasource/loader.ts

```diff
@@ -334,22 +331,13 @@ export async function changeLayerIndexPattern({
 }
 
 async function loadIndexPatternRefs(
-  savedObjectsClient: SavedObjectsClient
+  indexPatternsService: IndexPatternsService
 ): Promise<IndexPatternRef[]> {
-  const result = await savedObjectsClient.find<{ title: string }>({
-    type: 'index-pattern',
-    fields: ['title'],
-    perPage: 10000,
-  });
+  const indexPatterns = await indexPatternsService.getIdsWithTitle();
```

Comment:
I wonder whether we should pass in the `refresh` flag here (because that's the current behavior). I'm not sure whether the cache will be in a good state in all cases (like navigating to management, importing an index pattern, then coming back to Lens).

We already have the `isFullEditor` flag to prevent this on dashboards.

Reply:
Just moving here what we discussed offline. I did some tests, created an index pattern, removed one etc and it seems that it works fine so we decided to not do it for now. If we discover problems with the cache, I will enable it then